### PR TITLE
internal/blobstore: implement Open for multipart blobs

### PR DESCRIPTION
**Pull request by rogpeppe**
The only thing remaining is ListUpload.

### CODE REVIEW

**Review by fabricematrat — APPROVED** [review]

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2044/

File: internal/blobstore/multireader.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+func (r *multiReader) Close() error {
+	if r.r != nil {
+		return r.r.Close()
+	}
+	return nil
+}
+
+// Read implements io.Reader.Read.
+func (r *multiReader) Read(buf []byte) (int, error) {
+	logger.Infof("read %d bytes at %d, size %d", len(buf), r.pos, r.size)
```
Comment (mhilton) [inline internal/blobstore/multireader.go]: Info seems high here.

**Review by mhilton — APPROVED** [review]
LGTM

File: internal/blobstore/multireader.go (hunk 1 above)
Comment (rogpeppe) [reply internal/blobstore/multireader.go]: debugging remnant, removed.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2045/

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci-gce.jujugui.org:8080/job/charmstore-merge/561

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Pull request merged**